test/integration/shells_int_test.go

```diff
@@ -0,0 +1,99 @@
+package integration
+
+import (
+	"fmt"
+	"runtime"
+	"testing"
+
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/stretchr/testify/suite"
+)
+
+type ShellsIntegrationTestSuite struct {
+	tagsuite.Suite
+}
+
+func (suite *ShellsIntegrationTestSuite) TestShells() {
+	suite.OnlyRunForTags(tagsuite.Critical, tagsuite.Shell)
+
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	var shells []e2e.Shell
+	switch runtime.GOOS {
+	case "linux":
+		shells = []e2e.Shell{e2e.Bash, e2e.Fish}
+	case "darwin":
+		shells = []e2e.Shell{e2e.Bash, e2e.Zsh, e2e.Tcsh}
+	case "windows":
+		shells = []e2e.Shell{e2e.Cmd}
+	}
+
+	// Checkout the first instance. It doesn't matter which shell is used.
+	cp := ts.SpawnWithOpts(e2e.WithArgs("checkout", "ActiveState-CLI/small-python"))
```

Comment:
Suggest using `ACTIVESTATE_CLI_DISABLE_RUNTIME=false` here so the runtime checkout logic is centralized and isn't likely to interfere with the test assertions further down.